src/main/java/com/emc/ecs/zimbra/integration/EcsStoreManager.java

```diff
@@ -179,6 +179,33 @@ public InputStream readStreamFromStore(String locator, Mailbox mbox) throws IOEx
         return client.getObject(el.getBucketName(), el.getKey()).getObject();
     }
 
+    /**
+     * <p>
+     * Verify an object exists in the store.
+     * </p>
+     *
+     * @param locator identifier string for the blob as returned from write operation
+     * @param mbox    Mailbox which contains the blob
+     * @return boolean indicating whether the object exists
+     * @throws IOException
+     */
+    public boolean validateFromStore(String locator, Mailbox mbox) throws IOException {
+        EcsLogger.debug(String.format("validateFromStore() - start: locator - %s, accountId - %s", locator, mbox.getAccountId()));
+
+        EcsLocator el = EcsLocatorUtil.fromStringLocator(locator);
+
+        S3ObjectMetadata metadata = null;
+
+        try {
+            metadata = client.getObjectMetadata(el.getBucketName(), el.getKey());
+        } catch (Exception e) {
+            EcsLogger.error(String.format("Failed to retrieve metadata from - %s", el.getKey()));
+            throw new IOException(e);

```

Comment:
The code should check for at least a 404 response, in which case no exception should be thrown. There may be other responses as well for which no exception should be thrown.